### PR TITLE
Support credentials

### LINKED ISSUE #43: Add test for HTTP cookies

### PR DESCRIPTION
**Pull request by josh**
Incomplete.

I'm not sure how to support `credentials: 'omit'` with XHR.

Closes #43.

/cc @dgraham @annevk @mislav

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -196,3 +196,43 @@ promiseTest('supports HTTP DELETE', 2, function() {
     equal(request.data, '')
   })
 })
+
+promiseTest('doesnt send cookies with implicit omit credentials', 1, function() {
```
Comment (dgraham) [inline test/test.js]: "does not"

**Comment by dgraham** [comment]
We need to default to "omit" in the [Request constructor](https://github.com/github/fetch/blob/9b1ddc9ef47011c2c9b0c9bc89e7ec09cf5ca962/fetch.js#L136).

I wish the default was `same-origin` rather than `omit`. All of our fetch usage requires cookie headers.

**Comment by josh** [comment]
Any ideas how to disable cookies for `'omit'`? It seems like `withCredentials` is always ignored for same origin requests.

**Comment by annevk** [comment]
That feature does not exist in XMLHttpRequest. XMLHttpRequest only has same-origin and CORS.

I could be convinced to have the standard default to same-origin, but it seems rather ugly to have a different default from cross-origin. (Mistake with how we designed XMLHttpRequest...)

**Comment by josh** [comment]
@annevk it seems like the majority of fetch contexts (image, script, style, subresources) mostly use `"include"` even for cross origin resources with the exception of XHR's CORS semantics. I kinda agree that world would be a better place if stuff like `script` was actually `omit` by default. But for most direct `fetch()` api calls, I think I'll be using `same-origin` most the time.

**Comment by annevk** [comment]
Well they use include but also set mode to no-cors by default. If you use the crossorigin attribute I think the intent is omit and mode cors, though implementations might do same-origin and mode cors...

**Comment by dgraham** [comment]
> I could be convinced to have the standard default to same-origin

@annevk From my perspective, a default of `same-origin` is most useful. A default of `omit` means that `fetch` cannot be used easily by sites with user sessions maintained by cookie headers.

A simple `fetch('/some/url')` call becomes `fetch('/some/url', {credentials: 'same-origin'})` at *each* call site. I'd prefer to specify `omit` in the few cases it's needed instead.

For now, we've [added a default of `same-origin`](https://gist.github.com/dgraham/92e4c45da3707a3fe789#file-fetch-js-L37-L45) to the `$.fetch` wrapper we use on github.com.

If you think it's appropriate, maybe the default in the specification could be changed to `same-origin`?

**Comment by annevk** [comment]
I recommend filing a bug against the specification. I don't really like `same-origin` myself as what it does depends on the origin of the URL you passed in, which is super weird.

**Pull request merged**